### PR TITLE
[Core] Removed OrderItem based promotions

### PR DESCRIPTION
**Pull request by Zales0123**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Fixed tickets |
| License       | MIT
| Doc PR        |

In favor of upcoming changes with promotions system, it's time to clean up a little bit :) Here I remove some residues of ``OrderItem``-based promotions.

### CODE REVIEW

**Comment by michalmarcinkowski** [comment]
:+1:

File: UPGRADE.md [hunk 1]
```diff
@@ -223,6 +223,7 @@ class BookController extends ResourceController
 ### Core and CoreBundle
 
 * Moved ``taxCategory`` from ``Product`` to ``ProductVariant``
+* Set ``OrderItem`` not to be ``PromotionSubject``
```
Comment (pjedrzejewski) [inline UPGRADE.md]: Set not to be... :dancer: 

* ``OrderItem`` model from Core component no longer implements ``PromotionSubjectInterface``, remove all usages of ``addPromotion``, etc. from your custom code. Custom actions targeting items need to be adjusted, see ``ContainsProductRuleChecker`` for reference.

Comment (Zales0123) [inline UPGRADE.md]: I knew that it's not good enough :smile: :+1:

**Comment by rvanlaak** [comment]
For reference purpose:  https://github.com/Sylius/Sylius/pull/4954

**Pull request merged**